category/basic.lean

```diff
@@ -20,12 +20,26 @@ run_cmd mk_simp_attr `functor_norm
   g <$> (m <$> x) = (g ∘ m) <$> x :=
 (comp_map _ _ _).symm
 
-@[simp] theorem id_map' (x : f α) : (λa, a) <$> x = x := id_map _
+-- @[simp] theorem id_map' (x : f α) : (λa, a) <$> x = x := id_map _
```

Comment:
Why is it removed and commented out?

Reply:
I should bring it back in. There was a name clash with point-free lemmas in `category.functor`. Would `pf` be a useful suffix to add to my lemmas?

Reply:
I don't think so, since the terminology "point-free" is not used in lean in any official materials or in the community, so it's likely to just make people wonder what it stands for. Wouldn't your version be called `map_id`? I mean it's literally `map id = id`, right?

Reply:
That would make sense. It would be contrary to the naming in `is_lawful_functor` however.

Reply:
Actually I strongly dislike the names in `is_lawful_functor` anyway, I always get the order wrong. `map_comp` is used in other contexts way more than `comp_map`, but since it uses a notation the order is flipped. DX

Reply:
I agree, I think `comp_map` is not very useful as a rewrite rule. I wish the other way around was the basic law. That would make it easier to automate its proof and to use it.

Reply:
Actually, I'm wondering if `(λa, a) <$> x = x` shouldn't be renamed because it doesn't have `id` in it. `lam_self_map`, maybe?

Reply:
I like the name `id_map` or `map_id` with decorator because it's logically related to the real one, and it is what people will look for (probably after failing to rewrite with `id_map`).

Reply:
I stick to `id_map`, as this is what `is_lawful_functor` uses. The `'` at the end indicates that `id` is unfolded. I hope in later versions of the simplifier this will not be not necessary anymore.